metadata-service/factories/src/main/java/com/linkedin/metadata/boot/steps/RestoreGlossaryIndices.java

```diff
@@ -0,0 +1,213 @@
+  private final EntityRegistry _entityRegistry;
+
+  @Override
+  public String name() {
+    return this.getClass().getSimpleName();
+  }
+
+  @Nonnull
+  @Override
+  public ExecutionMode getExecutionMode() {
+    return ExecutionMode.BLOCKING;
+  }
+
+  @Override
+  public void execute() throws Exception {
+    log.info("Attempting to run RestoreGlossaryIndices upgrade..");
+    try {
+      if (_entityService.exists(GLOSSARY_UPGRADE_URN)) {
+        log.info("Glossary Upgrade has run before. Skipping");
+        return;
+      }
+
+      final AspectSpec termAspectSpec =
+          _entityRegistry.getEntitySpec(Constants.GLOSSARY_TERM_ENTITY_NAME).getAspectSpec(Constants.GLOSSARY_TERM_INFO_ASPECT_NAME);
+      final AspectSpec nodeAspectSpec =
+          _entityRegistry.getEntitySpec(Constants.GLOSSARY_NODE_ENTITY_NAME).getAspectSpec(Constants.GLOSSARY_NODE_INFO_ASPECT_NAME);
+      final AuditStamp auditStamp = new AuditStamp().setActor(Urn.createFromString(Constants.SYSTEM_ACTOR)).setTime(System.currentTimeMillis());
+
+      final DataHubUpgradeRequest upgradeRequest = new DataHubUpgradeRequest().setTimestampMs(System.currentTimeMillis()).setVersion(VERSION);
+      ingestUpgradeAspect(Constants.DATA_HUB_UPGRADE_REQUEST_ASPECT_NAME, upgradeRequest, auditStamp);
+
+      final int totalTermsCount = getAndRestoreTermAspectIndices(0, auditStamp, termAspectSpec);
+      int termsCount = BATCH_SIZE;
+      while (termsCount < totalTermsCount) {
+        getAndRestoreTermAspectIndices(termsCount, auditStamp, termAspectSpec);
+        termsCount += BATCH_SIZE;
+      }
+
+      final int totalNodesCount = getAndRestoreNodeAspectIndices(0, auditStamp, nodeAspectSpec);
+      int nodesCount = BATCH_SIZE;
+      while (nodesCount < totalNodesCount) {
+        getAndRestoreNodeAspectIndices(nodesCount, auditStamp, nodeAspectSpec);
+        nodesCount += BATCH_SIZE;
+      }
+
+      final DataHubUpgradeResult upgradeResult = new DataHubUpgradeResult().setTimestampMs(System.currentTimeMillis());
+      ingestUpgradeAspect(Constants.DATA_HUB_UPGRADE_RESULT_ASPECT_NAME, upgradeResult, auditStamp);
+
+      log.info("Successfully restored glossary index");
+    } catch (Exception e) {
+      log.error("Error when running the RestoreGlossaryIndices Bootstrap Step", e);
+      _entityService.deleteUrn(GLOSSARY_UPGRADE_URN);
+      throw new RuntimeException("Error when running the RestoreGlossaryIndices Bootstrap Step", e);
+    }
+  }
+
+  private void ingestUpgradeAspect(String aspectName, RecordTemplate aspect, AuditStamp auditStamp) {
+    final MetadataChangeProposal upgradeProposal = new MetadataChangeProposal();
+    upgradeProposal.setEntityUrn(GLOSSARY_UPGRADE_URN);
+    upgradeProposal.setEntityType(Constants.DATA_HUB_UPGRADE_ENTITY_NAME);
+    upgradeProposal.setAspectName(aspectName);
+    upgradeProposal.setAspect(GenericRecordUtils.serializeAspect(aspect));
+    upgradeProposal.setChangeType(ChangeType.UPSERT);
```

Comment:
This should be RESTATE right?

Reply:
NVM!

Reply:
yes definitely, got mixed up in the shuffle

Reply:
oh wait should it? This is when creating the two aspects, not when calling `produceMetadataChangeLog` (which I am using RESTATE